### ISSUE TITLE
'column not found' error if FK constraint references column that name has uppercase letters (MySQL)

### ISSUE THREAD

**Issue by myzkyy** [k1LoW/tbls#255]
### What happened

```
> CREATE DATABASE test;

> USE test;

> CREATE TABLE table1 (productCode INT PRIMARY KEY);

> CREATE TABLE table2 (productCode INT, FOREIGN KEY(`productCode`) REFERENCES table1(`productCode`));

> SHOW CREATE TABLE table1;
+--------+------------------------------------------------------------------------------------------------------------------------------+
| Table  | Create Table                                                                                                                 |
+--------+------------------------------------------------------------------------------------------------------------------------------+
| table1 | CREATE TABLE `table1` (
  `productCode` int(11) NOT NULL,
  PRIMARY KEY (`productCode`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+--------+------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

> SHOW CREATE TABLE table2;
+--------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table  | Create Table                                                                                                                                                                                                                         |
+--------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| table2 | CREATE TABLE `table2` (
  `productCode` int(11) DEFAULT NULL,
  KEY `productCode` (`productCode`),
  CONSTRAINT `table2_ibfk_1` FOREIGN KEY (`productCode`) REFERENCES `table1` (`productcode`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+--------+--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

Then, 

```
$ tbls doc mysql://user:password@127.0.0.1:3306/test ./dbdoc
not found column 'productcode' on table 'table1'
```

As you see, MySQL converts `productCode` into `productcode` in forein key constraint. This is OK in MySQL because
 column names are case-insensitive. However tbls compares column names in case-sensitive manner. This is the cause of the error.

### What you expected to happed

Column name comparison should be case-insensitive in MySQL.

### What stacktrace or error message from tbls did you see?

```
$ DEBUG=1 tbls doc mysql://root@127.0.0.1:3306/test ./dbdoc
not found column 'productcode' on table 'table1'
github.com/k1LoW/tbls/schema.(*Table).FindColumnByName
	/Users/k1low/src/github.com/k1LoW/tbls/schema/schema.go:149
github.com/k1LoW/tbls/drivers/mysql.(*Mysql).Analyze
	/Users/k1low/src/github.com/k1LoW/tbls/drivers/mysql/mysql.go:362
github.com/k1LoW/tbls/datasource.Analyze
	/Users/k1low/src/github.com/k1LoW/tbls/datasource/datasource.go:111
github.com/k1LoW/tbls/cmd.glob..func5
	/Users/k1low/src/github.com/k1LoW/tbls/cmd/doc.go:73
github.com/spf13/cobra.(*Command).execute
	/Users/k1low/go/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200710123203-207dc47664b4/command.go:846
github.com/spf13/cobra.(*Command).ExecuteC
	/Users/k1low/go/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200710123203-207dc47664b4/command.go:950
github.com/spf13/cobra.(*Command).Execute
	/Users/k1low/go/pkg/mod/github.com/spf13/cobra@v1.0.1-0.20200710123203-207dc47664b4/command.go:887
github.com/k1LoW/tbls/cmd.Execute
	/Users/k1low/src/github.com/k1LoW/tbls/cmd/root.go:184
main.main
	/Users/k1low/src/github.com/k1LoW/tbls/main.go:33
runtime.main
	/usr/local/Cellar/go/1.13.7/libexec/src/runtime/proc.go:203
runtime.goexit
	/usr/local/Cellar/go/1.13.7/libexec/src/runtime/asm_amd64.s:1357
```

### Anything else we need to know?

- In MySQL, database and table names are case-**sensitive** but columns are case-**insensitive**.
- In PostgreSQL, all quoted names are case-sensitive (if not quoted, case-insensitive).

### Environment

- tbls 1.43.1
- mysql  Ver 8.0.12 for osx10.13 on x86_64 (Homebrew)
- macOS Catalina 10.15.5
- zsh

**Comment by k1LoW** [comment]
Hi, @myzkyy Thank you for your report!!

I'll fix it as soon as possible.

**Comment by k1LoW** [comment]
I tried to check and could not reproduce it.

Is there something in the tbls.yml?

<details>

```
mysql> SELECT VERSION();
+-----------+
| VERSION() |
+-----------+
| 8.0.19    |
+-----------+
1 row in set (0.00 sec)

mysql> CREATE DATABASE `issue-255` DEFAULT CHARACTER SET utf8;
Query OK, 1 row affected, 1 warning (0.01 sec)

mysql> USE `issue-255`;
Database changed
mysql> CREATE TABLE table1 (productCode INT PRIMARY KEY);
Query OK, 0 rows affected (0.01 sec)

mysql> CREATE TABLE table2 (productCode INT, FOREIGN KEY(`productCode`) REFERENCES table1(`productcode`));
Query OK, 0 rows affected (0.03 sec)

mysql> SHOW CREATE TABLE table1;
+--------+--------------------------------------------------------------------------------------------------------------------------+
| Table  | Create Table                                                                                                             |
+--------+--------------------------------------------------------------------------------------------------------------------------+
| table1 | CREATE TABLE `table1` (
  `productCode` int NOT NULL,
  PRIMARY KEY (`productCode`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+--------+--------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> SHOW CREATE TABLE table2;
+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table  | Create Table                                                                                                                                                                                                                     |
+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| table2 | CREATE TABLE `table2` (
  `productCode` int DEFAULT NULL,
  KEY `productCode` (`productCode`),
  CONSTRAINT `table2_ibfk_1` FOREIGN KEY (`productCode`) REFERENCES `table1` (`productCode`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 |
+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql>
```

```console
$ tbls version
1.43.1
$ tbls doc my://root:mypass@localhost:33308/issue-255 -f issue/255 # no tbls.yml
issue/255/schema.svg
issue/255/table1.svg
issue/255/table2.svg
issue/255/README.md
issue/255/table1.md
issue/255/table2.md
```

</details>

**Comment by myzkyy** [comment]
@k1LoW 
Thank you for your response.

I noticed that it was a bug of MySQL and it has been fixed at 8.0.14.
https://dev.mysql.com/doc/relnotes/mysql/8.0/en/news-8-0-14.html
> Names of referenced columns of foreign keys were always shown in lowercase in SHOW CREATE TABLE output and the INFORMATION_SCHEMA.KEY_COLUMN_USAGE table. (Bug #27353767, Bug #88718)

Upgrading my local MySQL resolved this issue.

Sorry for bothering you. Thank you for your support!
